Register and login pages

zdwo opened the pull request.

juliadlutek commented on frontend/src/ui/pages/LoginPage/LoginPage.js:
```diff
@@ -0,0 +1,65 @@
+/* eslint-disable react/jsx-key */
+/* eslint-disable indent */
```
Please remove comments and fix the errors.

juliadlutek commented on frontend/src/ui/pages/LoginPage/LoginPage.scss:
```diff
@@ -0,0 +1,82 @@
+    border: 2px black solid;
+    &::placeholder {
+        font-family: var(--font-press-start-2p);
+        font-size: 20px;
```
Change all px units to rem

juliadlutek commented on frontend/src/ui/pages/LoginPage/LoginPage.scss:
```diff
@@ -0,0 +1,82 @@
+.btn {
+    border: none;
+    width: 100%;
+    // height: 124px;
```
Remove unnecessary comments

juliadlutek commented on frontend/src/ui/pages/RegisterPage/RegisterPage.js:
```diff
@@ -0,0 +1,128 @@
+/* eslint-disable react/jsx-key */
+/* eslint-disable indent */
```
Remove comments and fix the errors.

juliadlutek commented on frontend/src/ui/pages/RegisterPage/RegisterPage.js:
```diff
@@ -0,0 +1,128 @@
+            setErrors(['Server not available'])
+        } else {
+            if (res.status === 200) {
+                navigate('/login')
```
It seems that here something isn't working correctly. I created an account, but it didn't redirect me to another page; instead, the form cleared (the account was created successfully, but the navigation didn't work)

juliadlutek commented on frontend/src/ui/pages/RegisterPage/RegisterPage.scss:
```diff
@@ -0,0 +1,139 @@
+    align-items: center;
+    gap: 7rem;
+    justify-content: center;
+    // margin-top: -5rem;
```
Remove unnecessary comments

juliadlutek commented on frontend/src/ui/pages/RegisterPage/RegisterPage.scss:
```diff
@@ -0,0 +1,139 @@
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    width: 481px;
```
Change all px units to rem

juliadlutek commented on frontend/src/ui/pages/RegisterPage/RegisterPage.scss:
```diff
@@ -0,0 +1,139 @@
+    font-size: 10px;
+    // padding: 10px 0;
+    background-color: $button-start !important;
+    font-size: 1.3rem;
```
Many instances of the same CSS property are being overwritten within the same selector (e.g., font-size here). Could you please review the entire CSS files in this regard?

juliadlutek reviewed (changes requested):
Great job! The website looks very nice, and I love the pixel art. :) I've added a few comments, especially regarding the CSS files. Could you please go through them and address the points? The files seem a bit messy, and your attention to these details would be helpful.

juliadlutek commented:
Please create a mobile version - you can simply reduce the width of elements on the login page and reverse the flex direction for registration; i think that should look good :) Additionally, I noticed that if our page is intended to be non-scrollable on desktop, it would be nice to vertically center the elements. Currently, they are slightly closer to the top.

juliadlutek commented:
I also noticed that you adopted a different concept for character creation on the frontend compared to the backend. We'll need to modify the user model on the backend, replacing `outfitTop`/`outfitBottom` with `body` and `outfit`. I'll create a ticket for this, and I'll try to address it when I have some free time.

kbojarowska commented:
I think that it would also be good to have a redirect to login/main page from register, because if if by any chance we click register, than we have to go back by our browser. It would be good to do it by our page 
<img width="1918" alt="image" src="https://github.com/kbojarowska/DoTheBeast/assets/83237280/56d0770d-26de-4ab3-89b4-3df8e012b939">

zdwo commented:
- [x] unnecessary comments removed
- [x] errors fixed
- [x] redirect to login page after successfully creating the account is working now
- [x] links to redirect from register to login page and the pther way added
- [x] additionally added a popup informing the user that their account was successfully created

kbojarowska reviewed (changes requested):
> It seems that here something isn't working correctly. I created an account, but it didn't redirect me to another page; instead, the form cleared (the account was created successfully, but the navigation didn't work)

please resolve conflicts

juliadlutek commented:
> Please create a mobile version - you can simply reduce the width of elements on the login page and reverse the flex direction for registration; i think that should look good :) Additionally, I noticed that if our page is intended to be non-scrollable on desktop, it would be nice to vertically center the elements. Currently, they are slightly closer to the top.

Take a look also at this comment

kbojarowska commented:
<img width="958" alt="image" src="https://github.com/kbojarowska/DoTheBeast/assets/83237280/efb2455d-af71-4322-b862-594dca058ea6">
It isn't looking good on smaller screens

kbojarowska commented on backend/package.json:
```diff
@@ -31,4 +31,4 @@
     "swagger-jsdoc": "^6.2.8",
     "swagger-ui-express": "^5.0.0"
   }
-}
```
remove package json from changes

kbojarowska reviewed (changes requested):
Maybe it would be good to have this button in the end of form? (after creating avatar)
<img width="376" alt="image" src="https://github.com/kbojarowska/DoTheBeast/assets/83237280/3343a1b5-e118-400a-9609-4ca6b6c0091b">

Despite this 2 things everything is ok :)

zdwo commented:
> Maybe it would be good to have this button in the end of form? (after creating avatar)
I changed the styling and also updated the Button component (comment from another pr). Take a look if that's better :)

kbojarowska reviewed (approved):
Great job!
